File: loadercomplex-fabric/src/ver171/java/com/enderzombi102/loadercomplex/fabric17/impl/item/FabricItemStack.java
package com.enderzombi102.loadercomplex.fabric17.impl.item;

import com.enderzombi102.loadercomplex.api.minecraft.item.ItemStack;
import net.minecraft.text.LiteralText;
import org.jetbrains.annotations.NotNull;

public class FabricItemStack implements ItemStack {

	private final net.minecraft.item.ItemStack stack;

	public FabricItemStack(net.minecraft.item.ItemStack stack) {
		this.stack = stack;
	}

	public net.minecraft.item.@NotNull ItemStack getStack() {
		return stack;
	}

	@Override
	public String getDisplayName() {
		return this.stack.getName().getString();
	}

	@Override
	public void setDisplayName(String newName) {
		this.stack.setCustomName( new LiteralText( newName ) );
	}

	@Override
	public @NotNull Object getType() {
		return this.stack.getItem();
	}

	@Override
	public int getAmount() {
		return this.stack.getCount();
	}

	@Override
	public void setAmount(int amount) {
		this.stack.setCount(amount);
	}

	@Override
	public void setDamage(int damage) {
		this.stack.setDamage(damage);
	}

	@Override
	public int getDamage() {
		return this.stack.getDamage();
	}

	@Override
	public int getMaxDurability() {
		return this.stack.getMaxDamage();
	}

	@Override
	public int getMaxStackSize() {
		return this.stack.getMaxCount();
	}
}

File: loadercomplex-fabric/src/ver171/kotlin/com/enderzombi102/loadercomplex/fabric171/impl/item/FabricItemStack.kt
package com.enderzombi102.loadercomplex.fabric171.impl.item

import com.enderzombi102.loadercomplex.api.minecraft.item.ItemStack
import net.minecraft.text.LiteralText
import net.minecraft.text.Text
import net.minecraft.item.ItemStack as McItemStack


class FabricItemStack( private val stack: McItemStack ) : ItemStack {
	override fun getStack(): McItemStack =
		this.stack

	override fun getDisplayName(): String =
		Text.Serializer.toJson( stack.name )

	override fun setDisplayName( newName: String ) {
		stack.setCustomName( LiteralText( newName ) )
	}

	override fun getType(): Any =
		stack.item

	override fun getAmount(): Int =
		stack.count

	override fun setAmount( amount: Int ) {
		stack.count = amount
	}

	override fun setDamage( damage: Int ) {
		stack.damage = damage
	}

	override fun getDamage(): Int =
		stack.damage

	override fun getMaxDurability(): Int =
		stack.maxDamage

	override fun getMaxStackSize(): Int =
		stack.maxCount
}
